### PR TITLE
Move containerfile to root

### PR DESCRIPTION
**Pull request by yenaing-oo**
Regarding https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/issues/190

### CODE REVIEW

**Comment by yenaing-oo** [comment]
The app ran fine in development since static assets were being served directly from their original location without being minified. In production, however, Django couldn't seem to location `manage.py`:

image

I looked through the container's files and discovered that the **contents** of `./bagitobjecttransfer` were copied into `/app` rather than the directory iteself:
image

I think the Docker's `COPY` instruction copies the contents of a directory instead of the directory. Here's a similar [question ](https://stackoverflow.com/questions/26504846/copy-directory-to-another-directory-using-add-command)on Stackoverflow, but for the `ADD` instruction.

I changed the part in `Containerfile` copying sources files to copy files and directory separately, and this seemed to fix issue of `manage.py` not being found.

The next issue I ran into is that Django keeps trying to serve static assets from  `app/static` rather than `app/bagitobjecttransfer/static`. (I don't know how to recreate the error logs for this, I got another error instead when going back to this point).

I tried changing every possible setting in `base.py` to make it look for static files in the latter, but nothing seemed to work. The only thing that does work is changing `STATIC_ROOT` to the `app/static`.

File: bagitobjecttransfer/bagitobjecttransfer/settings/base.py [hunk 1]
```diff
@@ -151,7 +151,7 @@
 MEDIA_URL = '/media/'
 STATIC_URL = '/static/'
 MEDIA_ROOT = os.path.join(BASE_DIR, 'media')
-STATIC_ROOT = os.path.join(BASE_DIR, 'static/')
+STATIC_ROOT = os.path.join(BASE_DIR, '..', 'static')
```
Comment (danloveg) [inline bagitobjecttransfer/bagitobjecttransfer/settings/base.py:154]: Can you undo this change, and make the following change to `docker_prod.py`:

```python
# Overwrite base.py settings for media and static roots

MEDIA_ROOT = "/app/media"
STATIC_ROOT = "/app/static"
```

**Review by danloveg — CHANGES_REQUESTED** [review]
You're absolutely right that the static files need to be at `/app/static`, that was a good catch. I even [built that behaviour into the prod compose file](https://github.com/yenaing-oo/Secure-Record-Transfer/blob/move-containerfile-to-root/compose.prod.yml#L91), I must have just forgotten that I had made that decision.

We only need that change in the production configuration, hence why I've asked you to move that change to the base settings file to `docker_prod.py`.

File: bagitobjecttransfer/bagitobjecttransfer/settings/base.py (hunk 1 above)
Comment (yenaing-oo) [reply bagitobjecttransfer/bagitobjecttransfer/settings/base.py]: If we're keeping the path of the static directories fixed, would it make sense to use environment variables for this? We could perhaps use the same environment variable within `compose.prod.yml` and `ngix.conf`.

**Comment by yenaing-oo** [comment]
> 

Ah I think I found the other culprit for this behaviour in [ngix.conf](https://github.com/yenaing-oo/Secure-Record-Transfer/blob/1fc4438dd4d6a3f72ee3aa0fc8131f95bf82ff1c/docker/nginx/nginx.conf#L18-L20). It looks like this is telling the Ngix web server to serve static assets from `app/static`, whereas the line you mentioned in `compose.prod.yml` is defining a shared volume between Django and Ngix.

File: bagitobjecttransfer/bagitobjecttransfer/settings/base.py (hunk 1 above)
Comment (yenaing-oo) [reply bagitobjecttransfer/bagitobjecttransfer/settings/base.py]: > Can you undo this change, and make the following change to `docker_prod.py`:
> 
> ```python
> # Overwrite base.py settings for media and static roots
> 
> MEDIA_ROOT = "/app/media"
> STATIC_ROOT = "/app/static"
> ```

Resolved by: https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/192/commits/1fc4438dd4d6a3f72ee3aa0fc8131f95bf82ff1c and

**Review by danloveg — APPROVED** [review]
